### PR TITLE
Initial github setup

### PR DESCRIPTION
**Pull request by jlordiales**
- Added initial markdown files
- Added protobuf spec and generated Go types

### CODE REVIEW

File: protobuf/sparkplug_b.proto
Comment (serroba) [inline protobuf/sparkplug_b.proto:1]: Given this is a third party, do we need to commit this to our own project?

I'm wondering if we could treat this as a library dependency instead

File: protobuf/sparkplug_b.proto [hunk 1]
```diff
@@ -0,0 +1,224 @@
+// * Copyright (c) 2015, 2018 Cirrus Link Solutions and others
+// *
+// * This program and the accompanying materials are made available under the
+// * terms of the Eclipse Public License 2.0 which is available at
```
Comment (serroba) [inline protobuf/sparkplug_b.proto:4]: I don't know how this works. Can you commit to our Apache licensed software a different one?

File: protobuf/sparkplug_b.proto
Comment (jlordiales) [reply protobuf/sparkplug_b.proto:1]: I'm not aware of a way to include proto files as a library dependency

Comment (serroba) [reply protobuf/sparkplug_b.proto:1]: I had more the idea of doing what we do with the schemas in general. Having a dedicated place to just compile to different languages, in our case, go, and we import that library

Comment (serroba) [reply protobuf/sparkplug_b.proto:1]: So we separate the host library from the underlying proto generation

Comment (jlordiales) [reply protobuf/sparkplug_b.proto:1]: that feels like a lot of overhead considering the spec here is stable and won't change

Comment (serroba) [reply protobuf/sparkplug_b.proto:1]: I'm mostly concerned of having code that belongs to a different org embedded here.
It's not about design, it's more about correctness of licences.

Are you better informed on the legalities here?

Comment (serroba) [reply protobuf/sparkplug_b.proto:1]: https://chat.openai.com/share/c11ac26a-7686-42ce-b579-428c651fdbd2

I think we should be fine

**Review by serroba — APPROVED** [review]

**Pull request merged**